### PR TITLE
feat(core): new Convar txAdmin-suppressEvents

### PR DESCRIPTION
**Pull request by araynimax**
Implemented the feature-request of #595
This also fixes a missing event trigger of
'txAdmin:events:announcement' from
announcements created in the game menu

### CODE REVIEW

File: resource/menu/shared.lua [hunk 1]
```diff
@@ -26,6 +26,20 @@ function txPrint(...)
   print(msg)
 end
 
+local supressedEvents = GetConvar('txAdmin-supressEvents', '')
```
Comment (TasoOneAsia) [inline resource/menu/shared.lua]: Small typo
```suggestion
local supressedEvents = GetConvar('txAdmin-suppressEvents', '')
```

File: resource/menu/shared.lua [hunk 2]
```diff
@@ -26,6 +26,20 @@ function txPrint(...)
...
+-- Used to check if a specific event should be suppressed
+function isSuppressedEvent(eventName)
+  for eventToBypass in string.gmatch(supressedEvents, '[^,%s]+') do
+      if string.lower(eventName) == string.lower(eventToBypass) then
+          return true
+      end
+  end
+
+  return false
+end
```
Comment (TasoOneAsia) [inline resource/menu/shared.lua]: It is possible to slightly reduce time complexity by splitting and transforming the ConVar string into a map table. Then, once in the function, compare eventName against the string index in the map table. Reduces time complexity at the cost of space complexity.

If what I said makes no sense, don't worry about it, and keep it as is. This isn't a huge deal at all, and actual real-world use cases will see minimal difference. Change is up to you.

**Review by TasoOneAsia — COMMENTED** [review]
Thanks for the PR! Left a couple preliminary comments before @tabarra checks it out. I'd also like to see some documentation for this new ConVar. As it stands, we don't have a central place that documents all of the available txAdmin convars. The menu convars are located in the `docs/menu.md` file, but it may be beneficial to consolidate all of the convars into its own `ConVar.md` file or something of that sort. I'll let @tabarra comment on that direction.

File: resource/menu/shared.lua (hunk 2 above)
Comment (araynimax) [reply resource/menu/shared.lua]: Sounds good. I will adjust it that way.

File: resource/menu/shared.lua (hunk 1 above)
Comment (araynimax) [reply resource/menu/shared.lua]: 👍

**Comment by araynimax** [comment]
I added the convar documentation into the readme.md and docs/events.md. I think that are good places to find it as a dev.

**Comment by tabarra** [comment]
Hey @araynimax, thanks for the contribution!  
  
From my post a few months ago I changed my mind about suppressing/bypassing events.  
If you look at the `feat/core-playerlist` you will see that right now I'm doing a more `txAdmin:event`-centric approach to tx > fxserver communications.  
I even removed a few commands in favor of events:  
image
  
So i do not want the events to be bypassed, but this still brings the question, how do we handle people wanting to handle announcements/dm messages? I don't know, but that's much more a "ugly notistack bypass" than an event bypass per se.  
Also, considering we only have announcements and DM to overwrite, this could totally be a checkbox in the settings.  
  
That being said, I'll close this PR now and we can discuss how to better handle those cases back in #595 .

**Pull request closed without merging**